rotkehlchen/db/accounting_rules.py

```diff
@@ -248,8 +256,58 @@ def missing_accounting_rules(self, events: list[HistoryBaseEntry]) -> list[bool]
                         ),
                     )
                     missing_accounting_rule.append(cursor.fetchone()[0] == 0)
-                else:
-                    missing_accounting_rule.append(is_missing_rule)
+                    continue
+
+                # check the surrounding events to verify if this particular event will
+                # be processed by a previous one. First query the related events
+                related_events = history_db.get_history_events(
+                    cursor=cursor,
+                    filter_query=HistoryEventFilterQuery.make(event_identifiers=[event.event_identifier]),
+                    has_premium=True,
+                )
+                for related_event_idx, related_event in enumerate(related_events):
+                    counterparty = getattr(event, 'counterparty', None)
```

Comment:
This should be the for loop event right?

```suggestion
                    counterparty = getattr(related_event, 'counterparty', None)
```